components/win-users/src/account.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ptr::null_mut;
+use std::io::Error;
+
+use widestring::WideCString;
+use winapi::{LPCWSTR, BOOL, PSID, PSID_NAME_USE, LPDWORD, SID_NAME_USE};
+use winapi::winerror::*;
+
+use super::sid::Sid;
+
+extern "system" {
+  fn LookupAccountNameW(
+      lpSystemName: LPCWSTR, lpAccountName: LPCWSTR, Sid: PSID, 
+      cbSid: LPDWORD, ReferencedDomainName: LPCWSTR,
+      cchReferencedDomainName: LPDWORD, peUse: PSID_NAME_USE,
+  ) -> BOOL;
+}
+
+pub struct Account {
+  pub name: String,
+  pub system_name: Option<String>,
+  pub domain: String,
+  pub account_type: SID_NAME_USE,
+  pub sid: Sid
+}
+
+impl Account {
+  pub fn from_name(name: String) -> Option<Account> {
```

Comment:
You can get away with just accepting `&str` here instead of `String` because you're just going to call `clone()` on the param in `lookup_account()` down below